libs/langchain/langchain/text_splitter.py

```diff
@@ -1486,3 +1485,161 @@ def __init__(self, **kwargs: Any) -> None:
         """Initialize a LatexTextSplitter."""
         separators = self.get_separators_for_language(Language.LATEX)
         super().__init__(separators=separators, **kwargs)
+
+
+class HTMLSectionSplitter:
+    """
+    Splitting HTML files based on specified tag and font sizes.
+    Requires lxml package.
+    """
+
+    def __init__(
+        self,
+        headers_to_split_on: List[Tuple[str, str]],
+        xslt_path: str = "document_transformers/xsl/converting_to_header.xslt",
+        **kwargs: Any,
+    ) -> None:
+        """Create a new HTMLSectionSplitter.
+
+        Args:
+            headers_to_split_on: list of tuples of headers we want to track mapped to
+                (arbitrary) keys for metadata. Allowed header values: h1, h2, h3, h4,
+                h5, h6 e.g. [("h1", "Header 1"), ("h2", "Header 2"].
+            xslt_path: path to xslt file for document transformation.
+            Needed for html contents that using different format and layouts.
+        """
+        self.headers_to_split_on = dict(headers_to_split_on)
+        self.xslt_path = xslt_path
+        self.kwargs = kwargs
+
+    def split_documents(self, documents: Iterable[Document]) -> List[Document]:
+        """Split documents."""
+        texts, metadatas = [], []
+        for doc in documents:
+            texts.append(doc.page_content)
+            metadatas.append(doc.metadata)
+        results = self.create_documents(texts, metadatas=metadatas)
+
+        text_splitter = RecursiveCharacterTextSplitter(**self.kwargs)
+
+        return text_splitter.split_documents(results)
+
+    def split_text(self, text: str) -> List[Document]:
```

Comment:
it seems like the behavior of split documents and split text is different. Do we want this?

Reply:
no, Paul they are same but since the split_text does not have meta data, we do not have the title metadata. so we need to keep title